File: app/api/schemas/diagnostics.py
# app/api/schemas/diagnostics.py

from __future__ import annotations

from pydantic import BaseModel, Field


class NodeCountResponse(BaseModel):
    """Response model for a single node label count."""

    label: str = Field(..., description="Node label")
    count: int = Field(..., ge=0, description="Number of nodes with this label")


class NodesByLabelResponse(BaseModel):
    """Response model for GET /admin/diagnostics/nodes-by-label"""

    labels: list[NodeCountResponse] = Field(..., description="Node counts by label")


class RelCountResponse(BaseModel):
    """Response model for a single relationship type count."""

    type: str = Field(..., description="Relationship type")
    count: int = Field(..., ge=0, description="Number of relationships of this type")


class RelsByTypeResponse(BaseModel):
    """Response model for GET /admin/diagnostics/rels-by-type"""

    types: list[RelCountResponse] = Field(..., description="Relationship counts by type")

File: app/frontend/services/diagnostics.ts
import api from '@/lib/api';

export interface NodeCount {
    label: string;
    count: number;
}

export interface RelCount {
    type: string;
    count: number;
}

export interface NodesByLabelResponse {
    labels: NodeCount[];
}

export interface RelsByTypeResponse {
    types: RelCount[];
}

export interface Endpoint {
    method: string;
    path: string;
    name: string;
    tags: string[];
}

export interface EndpointsResponse {
    endpoints: Endpoint[];
}

export interface Metric {
    endpoint: string;
    count: number;
    avg_ms: number;
    min_ms: number;
    max_ms: number;
}

export interface MetricsResponse {
    metrics: Metric[];
}

export const diagnosticsService = {
    getNodesByLabel: async (): Promise<NodesByLabelResponse> => {
        const response = await api.get<NodesByLabelResponse>('/admin/diagnostics/nodes-by-label');
        return response.data;
    },

    getRelsByType: async (): Promise<RelsByTypeResponse> => {
        const response = await api.get<RelsByTypeResponse>('/admin/diagnostics/rels-by-type');
        return response.data;
    },

    getEndpoints: async (): Promise<EndpointsResponse> => {
        const response = await api.get<EndpointsResponse>('/admin/diagnostics/endpoints');
        return response.data;
    },

    getMetrics: async (): Promise<MetricsResponse> => {
        const response = await api.get<MetricsResponse>('/admin/diagnostics/metrics');
        return response.data;
    }
};
